src/gpu/intel/jit/conv/tiler.cpp

```diff
@@ -700,6 +701,23 @@ class conv_blocking_checker_t : public blocking_checker_t {
         return vec_ndims == 1;
     }
 
+    bool check_fp4_ok(const context_t &ctx) const {
+        auto prb = cfg_.prb();
+        if (utils::one_of(prb.src_data_type, data_type_t::dnnl_f4_e2m1,
+                    data_type_t::dnnl_f4_e3m0)
+                || utils::one_of(prb.wei_data_type, data_type_t::dnnl_f4_e2m1,
+                        data_type_t::dnnl_f4_e3m0)) {
+            for (auto &d : ctx.blk.iter())
+                for (auto t : input_tensors(cfg_.prb())) {
+                    if (tensor_conv_dim_index(d, t) == -1) continue;
+                    if (inner_stride(cfg_, t, d) == 1
+                            && inner_block(cfg_, d) < 8)
```

Comment:
```suggestion
                            && inner_block(cfg_, d) % 8)
```

You should also be able combine dense inner blocks to get to a multiple of 8. The check in `src/gpu/intel/jit/conv/config.cpp` may also need some adjustment. (Is it even necessary?)

Reply:
this works, thanks. The check in `config.cpp` is necessary since we have to reject shapes that wont generate valid tiling during pd creation.